apps/src/publicKeyCryptography/ModuloClock.jsx

```diff
@@ -56,32 +60,35 @@ const ModuloClock = React.createClass({
     const maximumTimePerSegment = Math.min(1000, maximumDuration / 10);
 
     this.targetDividend = dividend;
-    this.startTime = Date.now();
     this.interval = setInterval(this.tick, 33);
     this.onStep = onStep || function () {};
     this.onComplete = onComplete || function () {};
     this.duration = Math.min(maximumDuration, dividend * maximumTimePerSegment);
-    this.setState({currentDividend: 0});
+    this.setState({startTime: Date.now(), currentDividend: 0});
   },
 
   tick() {
-    const elapsedTime = Date.now() - this.startTime;
+
+    const elapsedTime = Date.now() - this.state.startTime;
     if (elapsedTime < this.duration) {
       const currentDividend = Math.floor(easeOutCircular(elapsedTime, 0, this.targetDividend, this.duration));
       this.onStep(currentDividend);
       this.setState({currentDividend});
-    } else {
+    } else if (this.state.currentDividend !== this.targetDividend) {
+      // Snap to final value, but there's one more step before the animation ends.
+      this.setState({currentDividend: this.targetDividend});
+    } else if (elapsedTime >= this.duration + FINALIZATION_DELAY) {
+      // End the animation once the final value has been shown for a moment.
       clearInterval(this.interval);
       this.onComplete(this.targetDividend);
-      this.setState({currentDividend: this.targetDividend});
       this.interval = null;
-      this.startTime = null;
       this.onStep = null;
       this.onComplete = null;
+      this.setState({startTime: null});
```

Comment:
Was actually necessary here to trigger a re-render when we go from the "final value reached but still show the pie filled" state to the "animation actually finished, start the fade."  I found that I could do the same by calling `setState` for `currentDividend` again but it felt hacky since that value didn't change.